### PR TITLE
fix: added cursor batch size in findall query to support document db

### PR DESCRIPTION
**Pull request by tkAppsmith**
#### PR fixes following issue(s)
Fixes #https://github.com/appsmithorg/appsmith-ee/issues/1659

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] Jest
- [ ] Cypress
>
>

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/BaseRepositoryImpl.java [hunk 1]
```diff
@@ -161,7 +161,9 @@ public Flux<T> findAll() {
                 .flatMapMany(principal -> {
                     Query query = new Query(notDeleted());
                     return mongoOperations.find(
-                            query, entityInformation.getJavaType(), entityInformation.getCollectionName());
+                            query.cursorBatchSize(10000),
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/BaseRepositoryImpl.java:164]: Which method is using the findAll like this? Can we instead override this method and fix usage? That way we will know exactly what type of usage is getting affected.

**Comment by tkAppsmith** [comment]
This is fixed but testing is blocked on a blocker that is also related to document db. https://github.com/appsmithorg/appsmith/issues/26090

**Comment by sharat87** [comment]
@tkAppsmith, do we not need this PR anymore?

**Pull request closed without merging**